plugins/modules/aci_l4l7_concrete_interface.py

```diff
@@ -0,0 +1,359 @@
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        tenant=dict(type='str', aliases=['tenant_name']),
+        device=dict(type='str'),
+        concrete_device=dict(type='str'),
+        state=dict(type='str', default='present',
+                   choices=['absent', 'present', 'query']),
+        concrete_interface=dict(type='str'),
+        pod_id=dict(type='int'),
+        node_id=dict(type='str'),
+        path_ep=dict(type='str'),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['tenant', 'device', 'concrete_device',
+                                 'concrete_interface']],
+            ['state', 'present', ['tenant', 'device', 'concrete_device',
+                                  'concrete_interface', 'pod_id', 'node_id',
+                                  'path_ep']]
+        ]
+    )
+
+    tenant = module.params.get('tenant')
+    state = module.params.get('state')
+    device = module.params.get('device')
+    concrete_device = module.params.get('concrete_device')
+    concrete_interface = module.params.get('concrete_interface')
+    pod_id = module.params.get('pod_id')
+    node_id = module.params.get('node_id')
+    path_ep = module.params.get('path_ep')
+
+    aci = ACIModule(module)
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class='fvTenant',
+            aci_rn='tn-{0}'.format(tenant),
+            module_object=tenant,
+            target_filter={'name': tenant},
+        ),
+        subclass_1=dict(
+            aci_class='vnsLDevVip',
+            aci_rn='lDevVip-{0}'.format(device),
+            module_object=device,
+            target_filter={'name': device},
+        ),
+        subclass_2=dict(
+            aci_class='vnsCDev',
+            aci_rn='cDev-{0}'.format(concrete_device),
+            module_object=concrete_device,
+            target_filter={'name': concrete_device},
+        ),
+        subclass_3=dict(
+            aci_class='vnsCIf',
+            aci_rn='cIf-[{0}]'.format(concrete_interface),
+            module_object=concrete_interface,
+            target_filter={'name': concrete_interface},
+        ),
+        child_classes=['vnsRsCIfPathAtt']
+    )
+
+    aci.get_existing()
+
+    if state == 'present':
+        if '-' in node_id:
+            path_type = 'protpaths'
+        else:
+            path_type = 'paths'
+
+        path_dn = ('topology/pod-{0}/{1}-{2}/pathep-[{3}]'.format(pod_id,
```

Comment:
should we include some form of validation of input to ensure valid path_dn is created?

Reply:
I'm not sure how much validation we can add, since the pathep for PCs and vPCs is just an IPG name (so just a generic string). The pod ID is already typed as int, and I've added a regex check to make sure the node ID is either an integer or a hyphen separated pair of integers.